src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -124,6 +131,9 @@ - (void)dealloc
 {
     // TODO: retain cycle and clean up https://github.com/project-chip/connectedhomeip/issues/34267
     MTR_LOG("MTRDevice dealloc: %p", self);
+
+    std::lock_guard lock(_lock);
```

Comment:
There shouldn't be a need to lock in dealloc

Reply:
Added a comment.